crates/ark/src/variables/variable.rs

```diff
@@ -918,132 +931,216 @@ impl PositronVariable {
-                                let index = path_element.parse::<isize>().unwrap();
-                                EnvironmentVariableNode::Concrete {
-                                    object: RObject::view(VECTOR_ELT(*object, index)),
-                                }
-                            },
+            // Actual promises, i.e. unevaluated promises can't be
+            // expanded in the variables pane so we would not get here.
 
-                            LISTSXP => {
-                                let mut pairlist = *object;
-                                let index = path_element.parse::<isize>().unwrap();
-                                for _i in 0..index {
-                                    pairlist = CDR(pairlist);
-                                }
-                                EnvironmentVariableNode::Concrete {
-                                    object: RObject::view(CAR(pairlist)),
-                                }
-                            },
+            let value = unsafe { PRVALUE(x) };
+            if r_is_unbound(value) {
+                x = unsafe { PRCODE(x) };
+            } else {
+                x = value;
+            }
+        }
 
-                            LGLSXP | RAWSXP | STRSXP | INTSXP | REALSXP | CPLXSXP => {
-                                if r_is_matrix(*object) {
-                                    EnvironmentVariableNode::Matrixcolumn {
-                                        object,
-                                        index: path_element.parse::<isize>().unwrap(),
-                                    }
-                                } else {
-                                    EnvironmentVariableNode::VectorElement {
-                                        object,
-                                        index: path_element.parse::<isize>().unwrap(),
-                                    }
-                                }
-                            },
+        Ok(EnvironmentVariableNode::Concrete {
+            object: RObject::view(x),
+        })
+    }
 
-                            _ => {
-                                return Err(harp::error::Error::InspectError { path: path.clone() })
-                            },
-                        }
-                    }
-                },
+    fn get_concrete_child_node(
+        object: RObject,
+        access_key: &String,
+    ) -> harp::Result<EnvironmentVariableNode> {
+        // Concrete nodes are objects that are treated as is. Accessing an element from them
+        // might result in special node types.
+
+        // First try to get child using a generic method
+        // When building the children list of nodes that use a custom `get_children` method, the access_key is
+        // formatted as "custom-{index}-{length(name)}-{name}". If the access_key has this format, we call the custom `get_child_at`,
+        // method, if there's one available:
+        let result = local!({
```

Comment:
We avoid `local!` since blocks in macros cause difficulties with formatting / assistance.

Reply:
Also I don't see a `?` so `local` is not needed?

Reply:
Fixed in https://github.com/posit-dev/ark/pull/561/commits/82fe4409c2be208c3b50f46a7a6d91e64d013062